lib/stimulus_reflex/reflex.rb

```diff
@@ -129,7 +130,21 @@ def params
     @_params ||= ActionController::Parameters.new(request.parameters)
   end
 
-  def dom_id(record_or_class, prefix = nil)
-    "#" + ActionView::RecordIdentifier.dom_id(record_or_class, prefix).to_s
+  def dom_id(record, prefix = nil, hash: "#")
+    hash + if record.is_a?(ActiveRecord::Relation)
+      [prefix, record.model_name.plural].compact.join("_")
+    elsif record.is_a?(ActiveRecord::Base)
+      ActionView::RecordIdentifier.dom_id(record, prefix).to_s
+    else
+      [prefix, record.to_s.sub(/^#/, "")].compact.join("_")
```

Comment:
I gotten into the habit of using `/\A/` vs `/^/` for matching the beginning of a string. 

- `^` Matches beginning of line
- `\A` Matches beginning of string

It probably doesn't matter here, but I almost never use `^` for this scenario.

Of course we could also use `String#squeeze`

```ruby
record.to_s.squeeze("#")
```

Reply:
🤦 would you believe I spent time looking at the String rdoc because I was *sure* there was a method that would do this, but I didn't connect the dots.

This is obviously superior.